### ISSUE TITLE
24.5 broke building on 64-bit ARM

### ISSUE THREAD

**Issue by brad0** [nzbgetcom/nzbget#468]
### NZBGet Version

v24.5-stable

### Platform

All

### Environment

```markdown
OpenBSD/arm64
```

### Problem or Question

I noticed a few commits making changes to compiler command line options. Something changed as it's now passing 32-bit ARM options on 64-bit ARM and breaking the build.

```
/home/brad/ports/pobj/nzbget-24.5/bin/cc -DHAVE_CONFIG_H -DPARPAR_ENABLE_HASHER_MD5CRC -DPARPAR_INVERT_SUPPORT -DPARPAR_SLIM_GF16 -D_DARWIN_C_SOURCE -D_DEFAULT_SOURCE -D_GNU_SOURCE -I/home/brad/ports/pobj/nzbget-24.5/build-aarch64/par2cmdline-turbo -I/home/brad/ports/pobj/nzbget-24.5/nzbget-24.5/par2cmdline-turbo/include -I/home/brad/ports/pobj/nzbget-24.5/nzbget-24.5/par2cmdline-turbo/parpar/gf16/gf16/opencl-include -O2 -pipe -DNDEBUG -std=c17 -fno-rtti -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -fstack-protector -Wno-c++98-compat -mfpu=neon -march=armv7-a -MD -MT par2cmdline-turbo/CMakeFiles/gf16.dir/parpar/gf16/gf16_shuffle_neon.c.o -MF par2cmdline-turbo/CMakeFiles/gf16.dir/parpar/gf16/gf16_shuffle_neon.c.o.d -o par2cmdline-turbo/CMakeFiles/gf16.dir/parpar/gf16/gf16_shuffle_neon.c.o -c /home/brad/ports/pobj/nzbget-24.5/nzbget-24.5/par2cmdline-turbo/parpar/gf16/gf16_shuffle_neon.c
cc: error: unsupported argument 'armv7-a' to option '-march='
ninja: build stopped: subcommand failed.
```

### Steps To Reproduce

Build.

**Comment by dnzbk** [comment]
Tested on my RPI aarch64 and it works fine.
Could you share the exact command you use to configure the build with CMake?
I'm not sure if this applies but we ran into [issue](https://github.com/nzbgetcom/nzbget/issues/460) with Ninja then we switched to using [ExternalProject](https://cmake.org/cmake/help/latest/module/ExternalProject.html) instead of [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) due to homebrew policy, 
Could you try using `Unix Makefiles` as your generator instead of `Ninja`?

**Comment by brad0** [comment]
> Tested on my RPI aarch64 and it works fine. Could you share the exact command you use to configure the build with CMake? I'm not sure if this applies but we ran into [issue](https://github.com/nzbgetcom/nzbget/issues/460) with Ninja then we switched to using [ExternalProject](https://cmake.org/cmake/help/latest/module/ExternalProject.html) instead of [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) due to homebrew policy, Could you try using `Unix Makefiles` as your generator instead of `Ninja`?

Yes, I did run into issue #460. Usually OpenBSD uses Ninja by default and we rarely run into issues nowadays. I am using the Makefile generator for now. But that wasn't the initial issue. I think a local CMake patch we have to workaround the par2cmdline-turbo dependency and providing the source code is introducing this breakage.

**Comment by dnzbk** [comment]
@brad0 
I'll definitely look into your patches, thank you.

Regarding [Sync par2cmdline-turbo](https://github.com/nzbgetcom/nzbget/issues/471):
I’ve released [NZBGet-testing](https://github.com/nzbgetcom/nzbget/releases/tag/testing) that includes [par2cmdline-turbo v1.2.0](https://github.com/animetosho/par2cmdline-turbo/releases/tag/v1.2.0). Unfortunately, [par2cmdline-turbo v1.2.0](https://github.com/animetosho/par2cmdline-turbo/releases/tag/v1.2.0) doesn’t include your patch. However, I’ve also synced the [nzbget branch](https://github.com/nzbgetcom/par2cmdline-turbo/tree/nzbget) of the fork with the most recent upstream [changes](https://github.com/animetosho/par2cmdline-turbo) including your patch.
So, as a temporary workaround, you can replace `GIT_TAG v1.2.0-nzbget` with `GIT_TAG nzbget` in the [par2-turbo.cmake](https://github.com/nzbgetcom/nzbget/blob/develop/cmake/par2-turbo.cmake) file and this should do the trick.
However, I'm not sure how to integrate this with your CMake patch.

**Comment by brad0** [comment]
The local patch was slightly modified to remove the foreach() loop and it seems to have resolved the issue.

https://cvsweb.openbsd.org/cgi-bin/cvsweb/~checkout~/ports/news/nzbget/patches/patch-cmake_par2-turbo_cmake?rev=1.3
